src/utils/atomWithStorage.ts

```diff
@@ -84,28 +89,32 @@ export function atomWithStorage<Value>(
   key: string,
   initialValue: Value,
   storage: AsyncStorage<Value> & { delayInit: true }
-): WritableAtom<Value, SetStateAction<Value> | typeof RESET, Promise<void>>
+): WritableAtom<
+  Value,
+  SetStateActionWithReset<Value> | typeof RESET,
```

Comment:
We don't need `typeof REST`, right?
```suggestion
  SetStateActionWithReset<Value>,
```

Reply:
ah, i just pushed out a commit to clean this up! https://github.com/pmndrs/jotai/pull/1346/commits/3ab023acb932910476889065816ba442cce7f6d2